pkg/cache/queue_info.go

```diff
@@ -350,6 +353,40 @@ func (qi *QueueInfo) MarkQueueForRemoval() {
 	}
 }
 
+// Return a copy of the properties for this queue
+func (qi *QueueInfo) GetProperties() map[string]string {
+	qi.Lock()
+	defer qi.Unlock()
+	props := make(map[string]string)
+	for key, value := range qi.properties {
+		props[key] = value
+	}
+	return props
+}
+
+// Set the properties that the unmanaged child queue inherits from the parent
+// only called on create of a new unmanaged queue
+// This currently only sets the sort policy as it is set on the parent
+// Further implementation is part of YUNIKORN-193
+func (qi *QueueInfo) setTemplateProperties() {
+	qi.Lock()
+	defer qi.Unlock()
+	// for a leaf queue pull out all values from the template and set each of them
+	// See YUNIKORN-193: for now just copy one attr from parent
+	if qi.isLeaf {
```

Comment:
It looks like this is limiting to a 2 level setup.
I wonder why it is not just inheriting the properties from its parent, no matter it is a leaf queue or not.

Reply:
The rest of the patch looks good. I think we can get this merged and unblock the new version. For more improvements, we can address in https://issues.apache.org/jira/browse/YUNIKORN-193